### PR TITLE
chore: bump `bb` to 0.43.0

### PR DESCRIPTION
**Pull request by TomAFrench**
# Description

## Problem\*

Resolves 

## Summary\*

This PR bumps the bb version to avoid stuff breaking on syncs

## Documentation\*

Check one:
- [ ] No documentation needed.
- [ ] Documentation included in this PR.
- [ ] **[For Experimental Features]** Documentation to be submitted in a separate PR.

# PR Checklist\*

- [ ] I have tested the changes locally.
- [ ] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

**Comment by Savio-Sou** [comment]
> Changes to circuit sizes

Circuit size optimization goes brrrrrr

File: tooling/noir_js_backend_barretenberg/src/backend.ts [hunk 1]
```diff
@@ -45,7 +45,7 @@ export class BarretenbergVerifierBackend implements VerifierBackend {
       const { Barretenberg, RawBuffer, Crs } = await import('@aztec/bb.js');
       const api = await Barretenberg.new(this.options);
 
-      const [_exact, _total, subgroupSize] = await api.acirGetCircuitSizes(this.acirUncompressedBytecode);
+      const [_exact, _total, subgroupSize] = await api.acirGetCircuitSizes(this.acirUncompressedBytecode, false);
```
Comment (michaeljklein) [inline tooling/noir_js_backend_barretenberg/src/backend.ts:48]:
```suggestion
      const honkRecursion = false;
      const [_exact, _total, subgroupSize] = await api.acirGetCircuitSizes(this.acirUncompressedBytecode, honkRecursion);
```

**Review by michaeljklein — APPROVED** [review]

**Comment by Savio-Sou** [comment]
Should we also bump the `bb` installation recommendation in docs as a part of / following this PR? https://noir-lang.org/docs/dev/getting_started/barretenberg/

**Comment by TomAFrench** [comment]
> Should we also bump the bb installation recommendation in docs as a part of / following this PR?

Sure, can you open a PR for this?

**Pull request merged**